@michaelkuhn, It hasn't worked for me.
I tested on mojave with the following clang
```
$ clang -v
Apple clang version 11.0.0 (clang-1100.0.33.17)
Target: x86_64-apple-darwin18.7.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

The error was
```
$ spack install libtirpc
==> Warning: clang@11.0.0-apple cannot build optimized binaries for "skylake". Using best target possible: "x86_64"
==> 40891: Installing libtirpc
==> Fetching https://sourceforge.net/projects/libtirpc/files/libtirpc/1.2.6/libtirpc-1.2.6.tar.bz2/download
######################################################################## 100.0%
==> Staging archive: /var/folders/ml/8jqjlf311qj53qyyvcl_50wc0000gn/T/myaccount/spack-stage-libtirpc-1.2.6-abt2qgwddz37qhpjkvjunlaqh2sfj6uh/download
==> Created stage in /var/folders/ml/8jqjlf311qj53qyyvcl_50wc0000gn/T/myaccount/spack-stage-libtirpc-1.2.6-abt2qgwddz37qhpjkvjunlaqh2sfj6uh
==> No patches needed for libtirpc
==> 40891: libtirpc: Building libtirpc [AutotoolsPackage]
==> 40891: libtirpc: Executing phase: 'autoreconf'
==> 40891: libtirpc: Executing phase: 'configure'
==> 40891: libtirpc: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j8'

28 errors found in build log:
     168      CC       libtirpc_la-bindresvport.lo
     169      CC       libtirpc_la-clnt_bcast.lo
     170      CC       libtirpc_la-clnt_dg.lo
     171      CC       libtirpc_la-clnt_generic.lo
     172    In file included from bindresvport.c:49:
     173    In file included from /var/folders/ml/8jqjlf311qj53qyyvcl_50wc0000gn/T/myaccount/spack-stage-libtirpc-1.2.6-abt2qgwddz37qhpjkvjunlaqh2sfj6uh/
            spack-src/tirpc/rpc/rpc.h:73:
  >> 174    /var/folders/ml/8jqjlf311qj53qyyvcl_50wc0000gn/T/myaccount/spack-stage-libtirpc-1.2.6-abt2qgwddz37qhpjkvjunlaqh2sfj6uh/spack-src/tirpc/rpc/rp
            cent.h:54:8: error: redefinition of 'rpcent'
     175    struct rpcent {
     176           ^
     177    /usr/include/netdb.h:159:8: note: previous definition is here
     178    struct rpcent {
     179           ^
     180    In file included from binddynport.c:41:
     181    In file included from /var/folders/ml/8jqjlf311qj53qyyvcl_50wc0000gn/T/myaccount/spack-stage-libtirpc-1.2.6-abt2qgwddz37qhpjkvjunlaqh2sfj6uh/
            spack-src/tirpc/rpc/rpc.h:73:
  >> 182    /var/folders/ml/8jqjlf311qj53qyyvcl_50wc0000gn/T/myaccount/spack-stage-libtirpc-1.2.6-abt2qgwddz37qhpjkvjunlaqh2sfj6uh/spack-src/tirpc/rpc/rp
            cent.h:54:8: error: redefinition of 'rpcent'
     183    struct rpcent {
...
```

If I run non parallel compilation, I get the following error
```
     163    Making all in src
     164      CC       libtirpc_la-auth_none.lo
  >> 165    auth_none.c:81:9: error: unknown type name 'mutex_t'
     166            extern mutex_t authnone_lock;
     167                   ^
     168    auth_none.c:83:2: warning: implicit declaration of function 'mutex_lock' is invalid in C99 [-Wimplicit-function-declaration]
     169            mutex_lock(&authnone_lock);
     170            ^
     171    auth_none.c:87:4: warning: implicit declaration of function 'mutex_unlock' is invalid in C99 [-Wimplicit-function-declaration]
     172                            mutex_unlock(&authnone_lock);
     173                            ^
     174    auth_none.c:103:2: warning: implicit declaration of function 'mutex_unlock' is invalid in C99 [-Wimplicit-function-declaration]
     175            mutex_unlock(&authnone_lock);
```

--- a/var/spack/repos/builtin/packages/libtirpc/package.py
+++ b/var/spack/repos/builtin/packages/libtirpc/package.py
@@ -13,12 +13,14 @@ class Libtirpc(AutotoolsPackage):
     homepage = "https://sourceforge.net/projects/libtirpc/"
     url      = "https://sourceforge.net/projects/libtirpc/files/libtirpc/1.1.4/libtirpc-1.1.4.tar.bz2/download"
 
+    version('1.2.6', sha256='4278e9a5181d5af9cd7885322fdecebc444f9a3da87c526e7d47f7a12a37d1cc')
     version('1.1.4', sha256='2ca529f02292e10c158562295a1ffd95d2ce8af97820e3534fe1b0e3aec7561d')
 
     provides('rpc')
 
     # FIXME: build error on macOS
     # auth_none.c:81:9: error: unknown type name 'mutex_t'
+    conflicts('platform=darwin', msg='Does not build on macOS')
 
     def configure_args(self):
         return ['--disable-gssapi']

